crates/api/src/builder/api.rs

```diff
@@ -679,42 +679,23 @@ where
             .await?;
 
         // Fetch constraints, and if available verify inclusion proofs and save them to cache
-        if let Some(constraints) = api.auctioneer.get_constraints(payload.slot()).await? {
-            let transactions_root: B256 = payload
-                .transactions()
-                .clone()
-                .hash_tree_root()?
-                .to_vec()
-                .as_slice()
-                .try_into()
-                .map_err(|e| {
-                    error!(error = %e, "failed to convert root to hash32");
-                    BuilderApiError::InternalError
-                })?;
-            let proofs = payload.proofs().ok_or(BuilderApiError::InclusionProofsNotFound)?;
-            let constraints_proofs: Vec<_> = constraints.iter().map(|c| &c.proof_data).collect();
-
-            verify_multiproofs(constraints_proofs.as_slice(), proofs, transactions_root).map_err(
-                |e| {
-                    error!(error = %e, "failed to verify inclusion proofs");
-                    BuilderApiError::InclusionProofVerificationFailed(e)
-                },
-            )?;
-
-            // Save inclusion proof to auctioneer.
-            api.save_inclusion_proof(
-                payload.slot(),
-                payload.proposer_public_key(),
-                payload.block_hash(),
-                proofs,
-                &request_id,
-            )
-            .await?;
-
-            info!(request_id = %request_id, head_slot, "inclusion proofs verified and saved to auctioneer");
+        let skip_inclusion_proof_verify_and_save = api
+            .relay_config
+            .constraints_api_config
+            .max_block_value_to_verify_wei
+            .map_or(false, |max_block_value_to_verify| payload.value() > max_block_value_to_verify);
+        if !skip_inclusion_proof_verify_and_save {
+            if let Err(err) = api.verify_and_save_inclusion_proofs(&payload, &request_id).await {
+                warn!(request_id = %request_id, error = %err, "failed to verify and save inclusion proofs");
+                return Err(err)
+            }
```

Comment:
What do you think of inverting the condition? Imo easier to read compared to a double negation
```rs
        let should_verify_and_save_inclusion_proofs = api
            .relay_config
            .constraints_api_config
            .max_block_value_to_verify_wei
            .map_or(true, |max_block_value_to_verify| payload.value() <= max_block_value_to_verify);
        if should_verify_and_save_inclusion_proofs {
            ...
         } else {
             info!("block value is greater than ...")
         }
```